nimble/controller/src/ble_ll_scan.c

```diff
@@ -1113,6 +1114,60 @@ ble_ll_scan_window_chk(struct ble_ll_scan_sm *scansm, uint32_t cputime)
     return 0;
 }
 #if MYNEWT_VAL(BLE_LL_CFG_FEAT_LL_EXT_ADV)
+static void
+ble_ll_scan_aux_data_free(struct ble_ll_aux_data *aux_scan)
+{
+    if (aux_scan) {
+        if (aux_scan->evt) {
+            ble_hci_trans_buf_free((uint8_t *)aux_scan->evt);
+            aux_scan->evt = NULL;
+        }
+        os_memblock_put(&ext_adv_pool, aux_scan);
+        STATS_INC(ble_ll_stats, aux_freed);
+    }
+}
+
+void
+ble_ll_scan_aux_data_ref(struct ble_ll_aux_data *aux_data)
+{
+    os_sr_t sr;
+
+    OS_ENTER_CRITICAL(sr);
+    if (!aux_data) {
+        OS_EXIT_CRITICAL(sr);
+        return;
+    }
+
+    aux_data->ref_cnt++;
+    STATS_INC(ble_ll_stats, aux_ref);
```

Comment:
tbh I do not see how such stat could be useful

Reply:
I agree this has limited usage but can be useful for debugging. When it is not changing when scanning ext advertisement, you know that you are out of aux_pool. Also when you set aux_pool to 1, it can help with debugging. However, I think that better would be to use system view for this, I will remove this stats and add system view calls instead.